### PR TITLE
[DO NOT MERGE] Bug 1727776, #1599 - Attempt at fixing Glean.js + iOS issues

### PR DESCRIPTION
**Pull request by brizental**
Related to: #1599

Hey folks! 

This is a "do not merge" PR because it contains an unreleased version of Glean.js. 

**tl;dr;** I was able to fix the issue for Release builds, but the issue persists in Debug builds. The current crash is caused by a bug in the Qt JS interpreter and I was not able to figure out a change in the JS code that does not cause it. Since it happens only in Debug mode, I would like to know from you if this is a good enough situation for now. I will file a bug against the Qt bug tracker in order to see if that gets fixed upstream, but that is a slow process for sure. If possible I would like for you to test it out as well to see if you also can't reproduce the bug in Release builds anymore.

There are many changes to Glean.js here -- including extremely verbose logs, those will be turned off in the real release. The [commit history](https://github.com/mozilla/glean.js/pull/744/commits) on the Glean.js PR has one by one the changes I made and why. On the description of that PR there is also a detailed description of my investigation efforts.

Regarding the VPN app changes, during my investigations I was working with the newest version of Glean.js, so I also updated Glean.js and glean_parser in here for testing. I also re-enabled Glean in all platforms, but that would probably need to be undone in order to disable it in iOS debug mode.

### CODE REVIEW

File: src/ui/main.qml [hunk 1]
```diff
@@ -63,33 +63,11 @@ Window {
             minimumWidth = Theme.desktopAppWidth
         }
 
-        Glean.initialize('MozillaVPN', VPNSettings.gleanEnabled && VPNFeatureList.get("glean").isSupported, {
+        Glean.setLogPings(true);
+        Glean.setDebugViewTag("vpn-testing");
```
Comment (brizental) [inline src/ui/main.qml:67]: This tag is [a debugging feature](https://mozilla.github.io/glean/book/reference/debug/debugViewTag.html). Setting this means all pings will be tagged upon sending, in practice that means the pings are re-routed to Glean's debug ping viewer. You can see the ping I sent when testing this here: https://debug-ping-preview.firebaseapp.com/pings/vpn-testing

**Comment by birdsarah** [comment]
EXCELLENT - I'll take a look

**Comment by brizental** [comment]
Quick update here. The crash I described has been confirmed as a Qt JS interpreter bug. See: https://bugreports.qt.io/browse/QTBUG-96788

**Comment by birdsarah** [comment]
Closing as this PR has served it's purpose and 0.21 of glean is released and we'll be working on integration. thanks so much @brizental .
